src/App.tsx

```diff
@@ -15,36 +16,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  DEFAULT = '',
+  ALPHABETICALLY = 'alphabetically',
+  LENGTH = 'length',
+}
+
+const getSortedGoods = (
+  goods: string[],
+  sortType: SortType,
+  reversed: boolean,
+): string[] => {
+  const sortedGoods = [...goods];
+
+  sortedGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABETICALLY:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reversed) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortValue, setActiveButton] = useState<SortType>(SortType.DEFAULT);
+  const [reversed, setReversed] = useState<boolean>(false);
+
+  const sortGoods = (type: SortType) => {
+    setActiveButton(type);
+  };
```

Comment:
[CODE KNOWLEDGE] - The function `sortGoods` should actually sort the goods, but it only sets the sort type. Consider renaming this function to reflect its purpose, such as `handleSortChange` or `updateSortType`, and move the sorting logic into this function if that's the intended behavior.